docs/user/knowledge-graph.rst

```diff
@@ -0,0 +1,37 @@
+.. _knowledge-graph:
+
+The Renku Knowledge Graph
+=========================
+
+The Knowledge Graph is the backbone of Renku. It is currently in beta and is
+used to record and query the relationships between data and code. In the future
+we will also use the Knowledge Graph to store and retrieve metadata and to
+provide search functionality. See the :ref:`lineage` section for a high-level
+description. 
+
+Generation of the Knowledge Graph
+---------------------------------
+
+Every time you use the ``renku run`` command we record some information about
+the inputs, the outputs, and the actual execution of your script. While you are
+working in your repository this information is simply stored in your project and
+accessible via tools like the ``renku log`` command. However, as soon as the
+project is pushed to the server, the commits are interpreted by the graph
+service and the data is fed into the database holding the Knowledge Graph. This
+happens automatically and there is no action required on your part, apart from 
+activating the Knowledge Graph integration.
+
+Private Projects and the Knowledge Graph
+----------------------------------------
+
+In short: if you enable the Knowledge Graph integration on a private project,
+the *metadata* about your project will become publicly accessible. Important:
+your *data* is in no way affected by this - the contents of your project are
+still private. What is metadata then? Things like filenames, paths within the
+project, people working on the project, project directory structure etc.  This
+is especially true if your private project references a public project - a
+lineage query will reveal the workflow steps taken within your private project
+using the public data. If sharing this level of metadata is not acceptable to
+you, then please do not turn on the Knowledge Graph integration at this stage.
```

Comment:
At the moment the user can choose to **not** activate the lineage for old projects, but this is not possible for new projects. This sentence is misleading since the user would probably think that he can always opt out

Reply:
can't opt out on new private projects?

Reply:
No. Should the user have this option?

Reply:
I assumed that was the case since it's opt-in for old projects as well.

Reply:
I believe we spoke about this and we decided to simply add a warning for private projects. My understanding was that KG is a main feature of Renku and it should be enabled, otherwise we may lose important info when lineage will be cross-projects.

Anyway adding an option to opt-out from the KG is easy, but the user experience for non-indexed projects should probably change, otherwise an annoying popup keeps showing every now and then during the navigation.